File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    def bfs(start):
        cnt = 0
        queue = deque([start])
        while queue:
            x = queue.popleft()
            for i in range(n):
                if computers[x][i] == 1 and not visit[i]:
                    visit[x] = True
                    queue.append(i)
                
    
    visit = [False] * n
    answer = 0
    for i in range(n):
        if not visit[i]:
            bfs(i)
            answer += 1
            
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static boolean[] nodes;
    public void dfs(int start, int[][] computers, int n){
        for(int i = 0; i < n; i++){
            if(i != start && computers[i][start] == 1){
                if(!nodes[i]){                
                    nodes[i] = true;
                    dfs(i, computers, n);
                }
            }
        }
    }
    public int solution(int n, int[][] computers) {
        nodes = new boolean[n];            
        int answer = 0;
        for(int i = 0; i < n; i++){
            if(!nodes[i]){
                answer++;
                nodes[i] = true;
                dfs(i, computers, n);
            }
                
        }
        return answer;
    }
}